### DISCUSSION TITLE
Path expansions for variables

### DISCUSSION THREAD

**Discussion by avidseeker** [jgm/pandoc#9812] [Q&A] [answered]
My goal is to specify a css file for revealjs given a generic path. The following defaults file works as expected
```yml
to: revealjs
variables:
  css: ['/home/myusername/.local/share/pandoc/revealjs.css']
```
However, when it is changed to
```yml
  css: ['~/.local/share/pandoc/revealjs.css']
```
pandoc gives a warning,
```log
[WARNING] Could not fetch resource ~/.local/share/pandoc/layouts/revealjs.css
```
But for some reason setting
```yml
  css: ['"$HOME"/.local/share/pandoc/layouts/revealjs.css']
```
results in pandoc *not* complaining about "could not fetch resource". Without a warning, the user expects pandoc to have found the file, but the resulting output is without style (possible bug? at least it should show a warning).

Now that none of this work, is it possible to at least have pandoc look for revealjs.css file within its `$XDG_DATA_HOME/pandoc$` directory? Currently this

```yml
  css: ['revealjs.css']
```

results in `[WARNING] Could not fetch resource revealjs.css`.

**Comment by iandol** [comment, marked as answer]
Did you try the syntax like:

```yaml
csl:  ${HOME}/mycsldir/special.csl
```

`${HOME}`, `${USERDATA}` and `${.}` are supported, though not in all defaults fields (only those that accept paths), but `css` should work...

https://pandoc.org/MANUAL.html#defaults-files

**Reply by avidseeker** [reply]
Unfortunately, this didn't work.
```
[WARNING] Could not fetch resource ${HOME}/.local/share/pandoc/layouts/revealjs.css
```
but  `/home/myusername/.local/share/pandoc/layouts/revealjs.css` works.

**Reply by iandol** [reply]
Hm, I just tested this:

```yaml
css: ${USERDATA}/templates/custom.css
```

becomes:

```html
<link rel="stylesheet" href="/home/cog5/.local/share/pandoc/templates/custom.css" />
```

I tested `${HOME}` and it worked too. Pandoc V3.2 on Ubuntu 22.04 -- are you using an ancient version of Pandoc?

**Reply by avidseeker** [reply]
I see now. Expansions don't work when `css:` is a key of `variables:`. i.e: the following
```
css: ${HOME}/.local/share/pandoc/layouts/revealjs.css
variables:
  css: ${HOME}/.local/share/pandoc/layouts/revealjs.css
```

results in
```
<link rel="stylesheet" href="${HOME}/.local/share/pandoc/layouts/revealjs.css"/>
<link rel="stylesheet" href="/home/myusername/.local/share/pandoc/layouts/revealjs.css"/>
```
